ui/client/components/graph/node-modal/editors/expression/Duration/PeriodEditor.tsx

```diff
@@ -20,31 +19,32 @@ type Props = {
   showValidation?: boolean,
   readOnly: boolean,
   isMarked: boolean,
+  components: Array<string>,
 }
 
 const SPEL_PERIOD_DECODE_REGEX = /^T\(java\.time\.Period\)\.parse\('(.*?)'\)$/
 const SPEL_PERIOD_SWITCHABLE_TO_REGEX = /^T\(java\.time\.Period\)\.parse\('P([0-9]{1,}Y)?([0-9]{1,}M)?([0-9]{1,}W)?([0-9]{1,}D)?'\)$/
 const SPEL_FORMATTED_PERIOD = (isoPeriod) => `T(java.time.Period).parse('${isoPeriod}')`
 const UNDEFINED_PERIOD = {
-  years: () => undefined,
-  months: () => undefined,
-  days: () => undefined
+  years: () => null,
+  months: () => null,
+  days: () => null,
 }
 
 export default function PeriodEditor(props: Props) {
 
-  const {expressionObj, onValueChange, validators, showValidation, readOnly, isMarked} = props
+  const {expressionObj, onValueChange, validators, showValidation, readOnly, isMarked, components} = props
 
   function isPeriodDefined(period: Period): boolean {
-    return period.years !== undefined || period.months !== undefined || period.days !== undefined
+    return period.years !== null || period.months !== null || period.days !== null
```

Comment:
do we need to know if these fields are null or undefined? maybe using js falsy is enough?

Reply:
Yes, they are null, which is set explicitly on fe

Reply:
But just to be sure i've changed it to !=